agent_shared/src/url_utils.rs

```diff
@@ -0,0 +1,82 @@
+pub trait AddFunctions {
```

Comment:
Perhaps this is a bit too generic of a name for this trait? I think using `UrlUtils` or something like that is a bit more explicit.